billing/views.py

```diff
@@ -46,11 +46,42 @@ def invoice_payment_succeeded(self, invoice: stripe.Invoice) -> None:
         owners: QuerySet[Owner] = Owner.objects.filter(
             stripe_customer_id=invoice.customer,
             stripe_subscription_id=invoice.subscription,
+            delinquent=True,
         )
-        owners.update(delinquent=False)
 
+        if not owners.exists():
+            return
+
+        admins = get_all_admins_for_owners(owners)
+        owners.update(delinquent=False)
         self._log_updated(list(owners))
 
+        # Send a success email to all admins
+
+        task_service = TaskService()
+        template_vars = {
+            "amount": invoice.total / 100,
+            "date": datetime.now().strftime("%B %-d, %Y"),
+            "cta_link": invoice.hosted_invoice_url,
+        }
+
+        for admin in admins:
+            if admin.email:
+                task_service.send_email(
```

Comment:
Can consider putting this in a try/catch with `continue` so if there is one bad email, it doesn't prevent the rest from going out

Reply:
@suejung-sentry as far as I understand it, the `send_email` call just creates a celery task that is later processed, totally async to this code. So, given that it shouldn't be possible for what you're describing to occur

Reply:
Oh nice! Missed that. Yeah looks good to go then!

Reply:
Thanks!